server/apis/v1/handler.go

```diff
@@ -1180,6 +1189,143 @@ func (h *handler) GetMonoVertexHealth(c *gin.Context) {
 	c.JSON(http.StatusOK, NewNumaflowAPIResponse(nil, response))
 }
 
+func formatArrayLabels(arrLabels []interface{}) string {
+	var labelSelectors []string
+	for _, value := range arrLabels {
+		labelSelectors = append(labelSelectors, fmt.Sprintf(`%s`, value))
+	}
+	return strings.Join(labelSelectors, ",")
+}
+
+func formatMapLabels(mapLabels map[string]interface{}) string {
+	var labelSelectors []string
+	for key, value := range mapLabels {
+		labelSelectors = append(labelSelectors, fmt.Sprintf(`%s="%s"`, key, value))
+	}
+	return strings.Join(labelSelectors, ",")
+}
+
+func (h *handler) GetMetricData(c *gin.Context) {
+
+	if h.prometheusClient == nil || h.prometheusClient.Client == nil {
+		h.respondWithError(c, "Prometheus client not set.")
+		return
+	}
+
+	var index int = -1
+	var fieldsMap = make(map[string]any)
+
+	var (
+		requestBody     map[string]any
+		startTimeString string
+		endTimeString   string
+		pattern         PatternData
+	)
+
+	if err := bindJson(c, &requestBody); err != nil {
+		h.respondWithError(c, fmt.Sprintf("Failed to decode JSON request body, %s", err.Error()))
+		return
+	}
+
+	// find index and pattern from config based on req pattern name
+	for i, p := range h.prometheusClient.ConfigData {
+		if p.Name == requestBody["name"] {
+			index = i
+			pattern = p
+			break
+		}
+	}
+
+	// throw error if pattern name not found in the config
+	if index == -1 {
```

Comment:
We can directly compare with` pattern != nil` right for the case it is not found. 
Index might not be required here, I don't see it getting further in the code. Is that correct?